dbms/src/Flash/Coprocessor/DAGRequest.cpp

```diff
@@ -169,6 +169,12 @@ const tipb::Executor & DAGRequest::rootExecutor() const
     return isTreeBased() ? dag_request->root_executor() : dag_request->executors(dag_request->executors_size() - 1);
 }
 
+String DAGRequest::rootExecutorID() const
+{
+    const auto & root_executor = rootExecutor();
+    return root_executor.executor_id();
```

Comment:
It may not be safe to assume that all executors have `executor_id`, if the request comes from TiSpark, it does not have the `executor_id`

Reply:
done. If no root executor_id, will use first executor to put ru consumption.

Reply:
see `checkOrSetExecutorId();`, executor id will not be nullptr in `DAGRequest`

Reply:
maybe just leave the code as it is as a guard.